### PR TITLE
android: add doma_kernel components

### PR DESCRIPTION
**Pull request by arminn**
This component should provide kernel image for the android build.

Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

### CODE REVIEW

File: prod-devel-rcar.yaml [hunk 1]
```diff
@@ -257,12 +257,32 @@ components:
...
+        dir: "."
+    builder:
+      type: android_kernel
+      env:
+        - "TARGET_BOARD_PLATFORM=%{SOC_FAMILY}"
+        - "BUILD_CONFIG=common/build.config.xenvm"
+        - "SKIP_MRPROPER=1"
+      target_images:
+        - "out/android12-5.4/common/arch/arm64/boot/Image"
+
```
Comment (lorc) [inline prod-devel-rcar.yaml:279]: Could you please add "out/android12-5.4/staging/vendor/lib/modules/pvrsrvkm.ko" as well? 
Just to be sure that Ninja will notice change in this file and trigger rebuild

Comment (arminn) [reply prod-devel-rcar.yaml:280]: Done

**Comment by lorc** [comment]
No we need someone to review this :) 
@rshim, maybe?

**Comment by rshym** [comment]
It's not clear why we need to upgrade the required version or moulin.
If this is an intentional change then maybe it's good to mention something like "pay attention that moulin 0.6 is required for this change". Else it's better to remove that change of version.
Despite that
`Acked-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>`

**Comment by lorc** [comment]
> If this is an intentional change then maybe it's good to mention something like "pay attention that moulin 0.6 is required for this change". Else it's better to remove that change of version.

Yes, this is intentional change, as "android_kernel" builder module is available only in 0.6.
But I agree that it should be mentioned in the commit message.

**Comment by arminn** [comment]
Added Acked-by

**Pull request merged**